src/contextify.cc

```diff
@@ -38,83 +38,85 @@ class ContextifyContext : public ObjectWrap {
     void Wrap(Handle<Object> handle);
 
     static void Init(Handle<Object> target) {
-        NanScope();
+        Nan::HandleScope scope;
 
-        Local<FunctionTemplate> ljsTmpl = NanNew<FunctionTemplate>(New);
+        Local<String> className = Nan::New("ContextifyContext").ToLocalChecked();
+
+        Local<FunctionTemplate> ljsTmpl = Nan::New<FunctionTemplate>(New);
         ljsTmpl->InstanceTemplate()->SetInternalFieldCount(1);
-        ljsTmpl->SetClassName(NanNew("ContextifyContext"));
-        NODE_SET_PROTOTYPE_METHOD(ljsTmpl, "run",       ContextifyContext::Run);
-        NODE_SET_PROTOTYPE_METHOD(ljsTmpl, "getGlobal", ContextifyContext::GetGlobal);
+        ljsTmpl->SetClassName(className);
+        Nan::SetPrototypeMethod(ljsTmpl, "run",       ContextifyContext::Run);
+        Nan::SetPrototypeMethod(ljsTmpl, "getGlobal", ContextifyContext::GetGlobal);
 
-        NanAssignPersistent(jsTmpl, ljsTmpl);
-        target->Set(NanNew("ContextifyContext"), ljsTmpl->GetFunction());
+        jsTmpl.Reset(ljsTmpl);
+        target->Set(className, ljsTmpl->GetFunction());
     }
 
     static NAN_METHOD(New) {
-        NanScope();
-
-        if (args.Length() < 1) {
-            NanThrowError("Wrong number of arguments passed to ContextifyContext constructor");
-            NanReturnUndefined();
+        if (info.Length() < 1) {
+            Nan::ThrowError("Wrong number of arguments passed to ContextifyContext constructor");
+            info.GetReturnValue().SetUndefined();
```

Comment:
The idiom is to do

return Nan::ThrowError(...);

As the function returns void.

On November 3, 2015 11:15:00 AM GMT+02:00, Ido Ben-Yair <notifications@github.com> wrote:
>>      }
>>  
>>      static NAN_METHOD(New) {
>> -        NanScope();
>> -
>> -        if (args.Length() < 1) {
>> -            NanThrowError("Wrong number of arguments passed to
>ContextifyContext constructor");
>> -            NanReturnUndefined();
>> +        if (info.Length() < 1) {
>> +            Nan::ThrowError("Wrong number of arguments passed to
>ContextifyContext constructor");
>> +            info.GetReturnValue().SetUndefined();
>
>Yeah, probably, but in this instance it's very easy to conform to the
>"standard", so I'll just change it.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/brianmcd/contextify/pull/190/files#r43727235